public/components/FeatureAnywhereContextMenu/AssociatedDetectors/components/__tests__/ConfirmUnlinkDetectorModal.test.tsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { findAllByTestId, render, waitFor } from '@testing-library/react';
+import { ConfirmUnlinkDetectorModal } from '../index';
+import { getRandomDetector } from '../../../../../../public/redux/reducers/__tests__/utils';
+import { DetectorListItem } from '../../../../../../public/models/interfaces';
+import userEvent from '@testing-library/user-event';
+
+describe('ConfirmUnlinkDetectorModal spec', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  const testDetectors = [
+    {
+      id: 'detectorId1',
+      name: 'test-detector-1',
+    },
+    {
+      id: 'detectorId2',
+      name: 'test-detector-2',
+    },
+  ] as DetectorListItem[];
+
+  const ConfirmUnlinkDetectorModalProps = {
+    detector: testDetectors[0],
+    onHide: jest.fn(),
+    onConfirm: jest.fn(),
+    onUnlinkDetector: jest.fn(),
+    isListLoading: false,
+  };
+
+  test('renders the component correctly', () => {
+    const { container, getByText } = render(
+      <ConfirmUnlinkDetectorModal {...ConfirmUnlinkDetectorModalProps} />
+    );
+    getByText('Remove association?');
+    getByText(
+      'Removing association unlinks test-detector-1 detector from the visualization but does not delete it. The detector association can be restored.'
+    );
+    expect(container).toMatchSnapshot();
+  });
+  test('should call onConfirm() when closing', async () => {
+    const { container, getByText, getByTestId } = render(
+      <ConfirmUnlinkDetectorModal {...ConfirmUnlinkDetectorModalProps} />
+    );
+    getByText('Remove association?');
+    await waitFor(() => {});
```

Comment:
What does this do exactly? Does this wait a specified amount of time?

Reply:
To be honest this shouldn't really add any time because the async function within is empty but this has been helpful in other tests where we execute some action and we want some wait as I assume just the waitFor function on its own adds some fractional wait time. However in this case since I am only really looking for the `toHaveBeenCalled()` to be true and not a different rendering I don't think its needed at all. The tests pass without it too.